common/src/test/kotlin/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketCoreImplTest.kt

```diff
@@ -0,0 +1,690 @@
+import io.homeassistant.companion.android.database.server.ServerSessionInfo
+import io.homeassistant.companion.android.database.server.ServerUserInfo
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.unmockkAll
+import io.mockk.verify
+import io.mockk.verifyOrder
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import okhttp3.OkHttpClient
+import okhttp3.WebSocket
+import okhttp3.WebSocketListener
+import okio.ByteString
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertSame
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertNotNull
+import org.junit.jupiter.api.assertNull
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
+import timber.log.Timber
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class WebSocketCoreImplTest {
+    private lateinit var mockConnection: WebSocket
+    private lateinit var webSocketCore: WebSocketCoreImpl
+    private lateinit var webSocketListener: WebSocketListener
+
+    @BeforeEach
+    fun setUp() {
+        Timber.plant(ConsoleLogTree)
+        ConsoleLogTree.verbose = true
+    }
+
+    @AfterEach
+    fun tearDown() {
+        unmockkAll()
+    }
+
+    private fun TestScope.setupServer(url: String = "https://io.ha", backgroundScope: CoroutineScope = this.backgroundScope) {
+        val mockOkHttpClient = mockk<OkHttpClient>(relaxed = true)
+        val mockServerManager = mockk<ServerManager>(relaxed = true)
+        val mockAuthenticationRepository = mockk<AuthenticationRepository>(relaxed = true)
+
+        val testServerId = 1
+        val testServer = Server(
+            id = testServerId,
+            _name = "Test Server",
+            connection = ServerConnectionInfo(
+                externalUrl = url,
+            ),
+            session = ServerSessionInfo(),
+            user = ServerUserInfo(),
+        )
+
+        mockConnection = mockk(relaxed = true)
+        every { mockServerManager.getServer(any<Int>()) } returns testServer
+        every { mockServerManager.authenticationRepository(testServerId) } returns mockAuthenticationRepository
+        coEvery { mockAuthenticationRepository.retrieveAccessToken() } returns "mock_access_token"
+        // The queue is a channel if we don't use a background scope the test hangs because the channel is never close
+        // In this configuration the channel will be close when the test will end
```

Comment:
Should this be moved to line 104, where you're setting up the `messageQueue`? I don't see anything queue-related here.